Sources/dikitgen/main.swift

```diff
@@ -1,21 +1,47 @@
 import Foundation
 import DIGenKit
 
-guard CommandLine.arguments.count == 2 else {
-    print("error: invalid arguments", to: &standardError)
-    print("usage: dikitgen <path to source code directory>", to: &standardError)
-    exit(1)
+enum Mode {
+    case version
+    case generate(path: String, excluding: [String])
 }
-
-if CommandLine.arguments[1] == "--version" {
-    print(Version.current)
-    exit(0)
+struct InvalidArgumentsError: Error {}
+func mode(from arguments: [String]) throws -> Mode {
+    switch arguments.count {
+    case 2 where arguments[1] == "--version":
+        return .version
+    case 2:
+        return .generate(path: arguments[1], excluding: [])
+    case 3..<Int.max:
+        let options = arguments.dropFirst(2)
+        guard
+            options.first == "--exclude" && options.count >= 2
+            else { throw InvalidArgumentsError() }
+        return .generate(path: arguments[1], excluding: options.dropFirst().map { $0 })
+    case _:
+        throw InvalidArgumentsError()
+    }
 }
 
-let path = CommandLine.arguments[1]
+let path: String
+let exclusions: [String]
+do {
+    switch try mode(from: CommandLine.arguments) {
+    case .version:
+        print(Version.current)
+        exit(0)
+    case .generate(let p, let xs):
+        path = p
+        exclusions = xs
+    }
+} catch {
+    print("error: invalid arguments", to: &standardError)
+    print("usage: dikitgen <path to source code directory> [--exclude <subpaths to exclude>...]", to: &standardError)
```

Comment:
Good documentation 👍 

[This part](https://github.com/ishkawa/DIKit/pull/36/files#diff-6788d7757fb482f02ec44b194fa189bcR18) of`mode(from:)` function seems to assume `--exclude DIR` appears only once .
If it is right, `...` doesn't express its behavior correctly.

Reply:
It accepts multiple directories (like `--exclude Example Tests`)
What stlye do you think is appropriate to represent it?

Reply:
Oh, I misunderstood this 🙇 

 I think it's better to specify `--exclude` for each exclusions like `--exclude DIR1 --exclude DIR2`, because the meaning is clearer especially when we have more options in the future.

Reply:
I got it. I’m going to fix to the style!